lib/spack/spack/cmd/find.py

```diff
@@ -175,6 +175,7 @@ def find(parser, args):
             tty.msg('No root specs')
         else:
             tty.msg('Root specs')
+            # TODO: Change this to not print extraneous deps and variants
```

Comment:
Ones that are matrixed to the spec as a constraint but are not valid for that spec. Like if you have a matrix of `[zlib, hdf5+mpi] X [^mvapich2, ^openmpi]`, the `^openmpi` or `^mvapich2` constraint on `zlib` is extraneous because it will be filtered out by the specialized stack concretization method.